capsules/core/src/spi_controller.rs

```diff
@@ -411,7 +411,7 @@ impl<'a, S: SpiMasterDevice<'a>> SpiMasterClient for Spi<'a, S> {
     ) {
         self.current_process.map(|process_id| {
             let _ = self.grants.enter(process_id, move |app, kernel_data| {
-                let rbuf = readbuf.map(|src| {
+                let rbuf = readbuf.inspect(|src| {
```

Comment:
Can you provide some context here?

I looked through the docs on `option::inspect()`, and it does look like a reasonable choice here. I guess the big difference is that this map just returned the original option, and `inspect()` does that by default. So probably a lint told us to use this function instead?

The bit that seems weird to me, is that we use `.map()` as a standard everywhere. And suddenly there's a brand new function with unclear semantics that any reader of code will have to look up before seeing that it's a slight optimization on just calling `.map()`.

Reply:
It's a new Clippy lint: https://rust-lang.github.io/rust-clippy/master/index.html#/manual_inspect

I think it's a reasonable lint, so between the choices of (a) turning it off or (b) changing our code to make Clippy happy, I chose the latter.

Sorry, I could have made this clearer in the PR description.

Reply:
`map` is still useful for when we ... well ... want to `map` things. Clippy complains because we're using `map` as a way to run a closure for a given `Option` / `Result` variant, but don't actually change the value or type stored in that container. Hence only changing those few occurrences.

Reply:
What I'm learning here is that to me `map()` means "let me run some code that accesses the internal value of the thing", and to you `map()` means map. I do think your view is the more reasonable one.

I think we should keep this change as it's the Rust thing to do. I do still feel like it's an unnecessary optimization on map, but it does exist and likely makes more sense to people who are thinking in a functional way.

Reply:
With the above I wasn't really trying to express my view on this, and honestly I think the distinction between `inspect` and `map` is somewhat moot -- I wouldn't have cared if Clippy didn't complain. This is just the "lowest-effort while still reasonable" way to get rid of warnings. :upside_down_face: